pennylane/operation.py

```diff
@@ -575,6 +585,28 @@ def decompose(self):
             return self.decomposition(wires=self.wires)
         return self.decomposition(*self.parameters, wires=self.wires)
 
+    @staticmethod
+    def compute_diagonalizing_gates(*params, wires, **hyperparams):
```

Comment:
We should have `compute_matrix` match this signature 💯 In fact, maybe all the `compute_*` should match?

Reply:
I think that is a good idea, if we decide to go with it!